### DISCUSSION TITLE
takeWhileP/takeWhileP1?

### DISCUSSION THREAD

**Discussion by Truedrog** [j-mie6/Parsley#175] [Q&A] [answered]
Right now I am subsituting them like this `many (satisfy f)`.
Or maybe I am missing something?

**Comment by j-mie6** [comment]
There is no _direct_ equivalent no (it's something I could consider adding), but you probably ~meant~ want `stringOfMany(satisfy(f))`: this is more like the equivalent of `many(satisfy(f)).map(_.mkString)` though, so it depends on if you wanted a `String` or `List[Char]` back.

**Reply by Truedrog** [reply]
I meant this https://hackage.haskell.org/package/megaparsec-9.3.0/docs/Text-Megaparsec.html#v:takeWhileP

But yeah, stringOfMany is good here. I was using many (satisfy f) because of that analogy

**Reply by j-mie6** [reply]
Yeah, I figured you did. It might be something I can consider adding, even if its definition is literally `def takeWhile(p: Char => Boolean) = stringOfMany(satisfy(p))`

**Comment by j-mie6** [comment, marked as answer]
Parsley `4.4.0-M2` has exposed an overloading of `stringOfMany`, and `stringOfSome`, that allow for a predicate to be passed: this models the desired behaviour of `takeWhileP` from `mgeaparsec`. This will make it into the `4.4.0` release properly in due course.
